docs/next/guides/building-and-tooling/modules.md

```diff
@@ -0,0 +1,1077 @@
+
+### Import core modules
+
+To import the core modules:
+1. Import Handsontable from `handsontable/base` (not from `handsontable`, which would give you the full bundle).
+2. If your bundler allows it, import Handsontable's CSS file as a whole (just like when you [install](@/guides/getting-started/installation.md) Handsontable).
+
+```js
+import Handsontable from 'handsontable/base';
+// if you bundler allows it, import Handsontable's CSS
+import 'handsontable/dist/handsontable.full.css';
+```
+
+Now, you're ready to use any [optional modules](#optional-modules) of your choice.
+
+## Optional modules
+
+Handsontable's optional modules are grouped into:
+- [Editor modules](#editor-modules)
+- [Renderer modules](#renderer-modules)
+- [Validator modules](#validator-modules)
+- [Cell type modules](#cell-type-modules)
+- [Plugin modules](#plugin-modules)
+- [Translation modules](#translation-modules)
+
+### Editor modules
+
+Editor modules contain Handsontable's [cell editors](@/guides/cell-functions/cell-editor.md).
+
+You can import the following editor modules:
+
+```js
+import {
+  registerEditor, // editors' registering function
+  AutocompleteEditor,
+  BaseEditor,
+  CheckboxEditor,
+  DateEditor,
+  DropdownEditor,
+  HandsontableEditor,
+  NumericEditor,
+  PasswordEditor,
+  SelectEditor,
+  TextEditor,
+} from 'handsontable/editors';
+```
+
+Each editor module contains a different type of cell editor:
+
+::: details Editor modules
+| Module               | Cell editor alias |
+| -------------------- | ----------------- |
+| `AutocompleteEditor` | `autocomplete`    |
+| `BaseEditor`         | `base`            |
+| `CheckboxEditor`     | `checkbox`        |
+| `DateEditor`         | `date`            |
+| `DropdownEditor`     | `dropdown`        |
+| `HandsontableEditor` | `handsontable`    |
+| `NumericEditor`      | `numeric`         |
+| `PasswordEditor`     | `password`        |
+| `SelectEditor`       | `select`          |
+| `TextEditor`         | `text`            |
+:::
+
+#### Import a single editor module
+
+To import a single editor module:
+
+1. Make sure you import the [core modules](#import-core-modules):
+    ```js
+    import Handsontable from 'handsontable/base';
+    // if you bundler allows it, import Handsontable's CSS
+    import 'handsontable/dist/handsontable.full.css';
+    ```
+2. Import the registering function and an editor module of your choice. For example:
+    ```js
+    import {
+      registerEditor,
+      PasswordEditor,
+    } from 'handsontable/editors';
+    ```
+3. Register your editor module, to let Handsontable recognize it. For example:
+    ```js
+    registerEditor(PasswordEditor);
+    ```
+
+A full example:
+
+```js
+import Handsontable from 'handsontable/base';
+import 'handsontable/dist/handsontable.full.css';
+import {
+  registerEditor,
+  PasswordEditor,
+} from 'handsontable/editors';
+
+registerEditor(PasswordEditor);
+```
+
+#### Import all editor modules
```

Comment:
The sections:

- Import all renderer modules
- Import all editor modules
- Import all validator modules
- Import all cell type modules
- Import all plugin modules

Explain our convenience APIs that diminish the purpose of using modules in the first place. The lengthly descriptions of these APIs significantly water down the content of this page and undoubtedly make it confusing for the reader (what are they trying to tell me with that?)

My advice is to remove these sections and just add a mention of these convenience APIs clearly in the the section "List of all module imports".

Reply:
@warpech - done: [7fa2122](https://github.com/handsontable/handsontable/pull/9059/commits/7fa21224300a26779d485e3fd2835cd264cc54f8)

Reply:
It seems that now the information about the convenience methods is now only available in the code snippet below "List of all module imports". This is a little bit too obscure. 

Could the text before the snippet be extended to have a short description or the convenience methods `registerAllEditors`, `registerAllRenderers`, `registerAllValidators`, `registerAllCellTypes`, `registerAllPlugins`, `registerAllModules`. It is needed given the fact that these methods do not even have an entry in the API reference. We shouldn't just throw them unannounced inside the snippet.

Reply:
Actually, I think it would be more correct to have 2 snippets below "List of all module imports":

1. One that uses the convenience methods (without the tutorial steps, just a description and the snippet)
2. One that registers all modules explicitly (without the tutorial steps, just a description and the snippet)

Reply:
@warpech - done: [affb722](https://github.com/handsontable/handsontable/pull/9059/commits/affb7226e03707073cd09b76468ba5f55c60dc5b)